core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -578,16 +584,19 @@ class LogManager(logDirs: Seq[File],
   }
 
   /**
-   * Make a checkpoint for all logs in provided directory.
+   * Make a checkpoint for all logs in provided directory by default.
+   * The affectedLog argument is an optimization that can be used if we only want to checkpoint a few partitions.
    */
-  private def checkpointLogRecoveryOffsetsInDir(dir: File): Unit = {
+  private def checkpointLogRecoveryOffsetsInDir(dir: File, affectedLog: Option[Seq[Log]] = None): Unit = {
     for {
       partitionToLog <- logsByDir.get(dir.getAbsolutePath)
       checkpoint <- recoveryPointCheckpoints.get(dir)
     } {
+      checkpoint.write(partitionToLog.mapValues(_.recoveryPoint))
       try {
-        checkpoint.write(partitionToLog.mapValues(_.recoveryPoint))
-        allLogs.foreach(_.deleteSnapshotsAfterRecoveryPointCheckpoint())
+        affectedLog.getOrElse(partitionToLog.values)
+          .filterNot(log => log.dir.getAbsolutePath.equals(dir.getAbsolutePath)) // it's safe to filter out the partitions with the incorrect directory
```

Comment:
It seems that log.dir.getAbsolutePath should be log.dir.getParentFile.getAbsolutePath? Could we make this method visible at the package level and add a unit test?